### ISSUE TITLE
animint2 Reference Website

### ISSUE THREAD

**Issue by ampurr** [animint/animint2#97]
Here's the thread that will track the development of the `animint2` reference website. 🐈

**Comment by ampurr** [comment]
@tdhock, sorry—I'm gonna ask you to do something for me. I would do it myself, but I don't have security clearance (or whatever it's called). 😅

Could you go to `Settings > Pages` and deploy the GitHub Pages site from the [`reference-website-yay` branch](https://github.com/tdhock/animint2/tree/reference-website-yay)? Later on we'll probably start deploying from the main branch and automate things with GitHub Actions (unless you don't want to).

But for now, I just wanna set up a preliminary site in your repo instead of my fork. Also, `usethis::use_pkgdown_github_pages()` doesn't work unless GitHub Pages is set up.

Thanks! As always, no rush. :>

**Comment by tdhock** [comment]
I did that, does it work for you?

**Comment by ampurr** [comment]
Thanks! Lemme run the `pkgdown` stuff and play around. I'll you know as soon as I do plus a delay of perhaps up to several minutes, since instantaneous communication is impossible. :>

**Comment by ampurr** [comment]
Hmm. When you deployed the site, in

`Settings > Pages > Build and deployment > Source`,

are you using GitHub Actions or are you deploying from a branch?

**Comment by tdhock** [comment]
It says Deploy from a branch, is that ok? Can I give you access some how so you can do this config?

**Comment by ampurr** [comment]
Yeah, deploy from a branch should be right. Weird that it works from my fork but throws up a `GitHub API error (404): Not Found` error when I do it from my branch in this repo. I'll figure it out.

Feel free to give me access, and I'll happily return it when the GitHub Pages site is set up. But I'm not sure how to do it. Maybe give me [admin powers](https://stackoverflow.com/questions/40673052/how-to-promote-a-collaborator-to-have-owner-rights)? I can't find anything about whether admins have power over setting up Pages, though. If you have more modular control, feel free to give me access only to the Pages settings or something. But I haven't been able to find anything about that.

**Comment by ampurr** [comment]
> Yeah, deploy from a branch should be right. Weird that it works from my fork but throws up a `GitHub API error (404): Not Found` error when I do it from my branch in this repo. I'll figure it out.

Hypothesis: I'm using _my_ personal access token, which maybe probably doesn't work for this repo since it's not mine. I may need you to generate a token for this repo and send it to me.

When I started this project, I had no idea it would require so many tedious little tasks from you, which I apologize for. Do you think I should continue this stuff from my fork instead? Of course, that has other problems and probably increases your future workload in exchange for a reduced workload now. :thinking:

**Comment by tdhock** [comment]
OK thanks for the info. I'm not sure how to give you admin access, because I think github changed their web site since the last time I did that. I don't know if this will help, but I saw a message saying "create an organization to grant access controls" so I created the animint organization, and invited you.

**Comment by ampurr** [comment]
Thanks! I joined but the animint  organization doesn’t seem to have access to any repositories. I don’t know if it’s because the `animint2` repo wasn’t added, or if it’s cuz I’m missing some sort of permission. Either way: I don’t get access to the settings for `animint2` and therefore can’t control the stuff about pages.

Don’t worry about it, though. If it’s okay with you: I can just do stuff on my fork and at some later date we can have an electron meeting and figure it out this GUI stuff together. And I can just merge my fork into a branch here later, too.

**Comment by tdhock** [comment]
hi @ampurr  and @Faye-yufan I gave you admin access to this repo (recently transferred to animint/animint2 from tdhock/animint2) does that work for you?

**Comment by ampurr** [comment]
@tdhock: It works! Thank you very much. :>

**Comment by Faye-yufan** [comment]
Hi @tdhock , yes it works, thank you!

**Comment by ampurr** [comment]
Status update: I'm currently working on editing `README.md`, which will serve as the home page for the animint2 reference site, and which @tdhock suggested that I edit during our last meeting. (It is currently a mirror image of `README.org`.)

If there's anything I should really make sure to include or exclude, please let me know. 🐱

**Comment by tdhock** [comment]
for sure include screenshots, installation, and basic usage example. anything else you want to delete please move to a wiki page.

**Comment by ampurr** [comment]
Okay, will do! What do you mean by "anything else you want to delete please move to a wiki page"?

**Comment by tdhock** [comment]
There is a wiki on this github repo, https://github.com/animint/animint2/wiki and you could move the README section "Differences with Other Packages" to a new wiki page with that name.

**Comment by ampurr** [comment]
Sorry, I'm still confused. My fault, not yours. But I took some time to think about it and I'm less confused than before.

So I'm going to try and say what I'm doing. And then I'm going to say some possible interpretations of what you're requesting. Then you can let me know if any of the interpretations are correct.

I generated `README.md` from `README.org`. I'm editing `README.md`, which does not affect `README.org`. My goal is to have the animint2 reference site homepage look different from the GitHub README. GitHub will read from the Org file, while the website will get its homepage from the Markdown file. (I'll rename `README.md` to `index.md` to prevent a conflict.)

I'm confused partly cuz I'm not deleting anything from `README.org`, so it's unclear what the benefit of adding parts of it to the animint2 wiki are.

Some interpretations:

1. You want the Org and Markdown READMEs to be in sync. That's why I need to move some deleted parts to the wiki.
2. The Org and Markdown READMEs must be in sync. You want the information present in the Org file but excluded from the Markdown file to be part of the wiki.
3. My status update was unclear, and your request is based on an assumption that I'm unaware of.

**Comment by tdhock** [comment]
yes for 1, and actually just delete the org file in your PR please.

**Comment by ampurr** [comment]
Understood. Will do. Thanks for clarifying. :>

**Comment by ampurr** [comment]
Today I learned that I do not know how GitHub's milestone thingy works.

**Comment by ampurr** [comment]
Status update: added the "Differences with other packages" section from `README.org` to the animint2 repo wiki. I'll be linking that wiki page on the animint2 reference homepage.

No plans to add anything else to the wiki.

**Comment by ampurr** [comment]
Minor status update: I'm unable to include basic usage examples in the README. Why? GitHub doesn't support JavaScript in its READMEs, and animints are made of sugar and spice and also JavaScript. I'm moving the examples to a vignette and including a link in the Use section.

**Comment by ampurr** [comment]
Status update: work on the examples vignette is going all right. The biggest difficulty is selecting a built-in dataset that shows off `animint2`'s features. I've tried `morley` and `Theoph`, but they're too sparse to display the faceted graphs well. Will probably try `msleep` next.

**Comment by ampurr** [comment]
> Status update: work on the examples vignette is going all right. The biggest difficulty is selecting a built-in dataset that shows off `animint2`'s features. I've tried `morley` and `Theoph`, but they're too sparse to display the faceted graphs well. Will probably try `msleep` next.

Just simulate the data. You’re demonstrating some package features, not doing science. 😛

**Comment by ampurr** [comment]
Status update: what was meant to be a brief short examples section has expanded into a quick start guide for `animint2`. The simulated data idea has worked out, though it took some time to shape the data so that they aren't too random or too perfect.

The most difficult aspect of the quick start guide is explaining something like the grammar of graphics without actually relying on the grammar itself. Mostly cuz I'm worried the grammar adds too much unneeded complexity for something that a new user's meant to understand in, like, 10 minutes.

I'm unsure of this decision. It's possible that _not_ explaining the grammar is a bad idea. Advice is welcome—no pressure if you don't have any.

**Comment by tdhock** [comment]
sounds good to me

**Comment by ampurr** [comment]
Okay, cool. :>

**Comment by tdhock** [comment]
can you share a link to your current draft reference website / quick start guide?

**Comment by ampurr** [comment]
Would you like me to commit what I currently have in a branch and send you the link?

The stuff I've committed so far is in the [reference-website-yay branch](https://github.com/animint/animint2/tree/reference-website-yay). I haven't committed the current `README.md` and `quick-start.Rmd` (and `website-setup.Rmd`) files yet, though. Why? They're either in various states of incompleteness or dissatisfactory-ness.

**Comment by ampurr** [comment]
Status update: I've completed my draft for the [quick start guide](https://github.com/animint/animint2/blob/reference-website-yay/vignettes/quick-start.Rmd). Lemme know what you think. Harsh criticism is welcome. There will likely be some intense editing in the future regardless.

I spent a lot of time experimenting with `<table>` and `display: flex;` to try and hack together a `ggplot2`-style facet, but that didn't work out, and there's no trace of it in the current draft.

I wanna say that I anticipate a working website by the end of the week. But I've been terrible at estimating how long things will take in this project. 🐈

**Comment by ampurr** [comment]
Oh, I forgot to mention. Something I wanna do is to explain how to use `clickSelects`. (Currently, only `showSelected` is explicitly used.) Part of the pedagogical difficulty is:

1. There's already an implicit `clickSelects` active when you use `animint()`.
2. Explicitly declaring `clickSelects` doesn't seem to be useful unless your data are super dense, like with the World Bank data.

**Comment by ampurr** [comment]
Status update: the new `README.md` is done. :>

**Comment by ampurr** [comment]
Status update: `animint2`'s functions are being reorganized in the `_pkgdown.yaml` file. 🧹

**Comment by ampurr** [comment]
Status update: `animint2`'s functions have been reorganized. Currently, `pkgdown::build_site()` throws up an error,

```
Error in yaml.load(readLines(con, warn = readLines.warn), error.label = error.label,  : 
(./_pkgdown.yml) Parser error: while parsing a block mapping at line 9, column 1 did not find expected key at line 53, column 1
```

which has not previously shown up. So I probably messed up the syntax somewhere. Trying to figure it out. :>

**Comment by ampurr** [comment]
### Status Update

Skimming through the [YAML specifications](https://yaml.org/spec/1.2.2/), I learned that whitespace is syntactically significant in YAML files... except when it's not? Cool cool cool.

Anyway, I figured out the parser error. It was a typo. `-title: "Adjustments"` should have been `- title: "Adjustments"`. Whoops. There's that significant whitespace. 😅

Using `check_pkgdown()` throws up new errors:

```
Error in `map2()`:
ℹ In index: 2.
Caused by error in `purrr::map()`:
ℹ In index: 4.
Caused by error:
! In '_pkgdown.yml', topic must be a known selector function
✖ Not 'gg-add'
Caused by error in `gg - add`:
! unused argument (add)
Run `rlang::last_trace()` to see where the error occurred.
```

Which is progress! :>

**Comment by ampurr** [comment]
Status update: Resolved the new errors. Three of the topics were under the wrong name. So I fixed that. But it also turns out that I missed about 50 topics, so I'm gonna add them. Figured I'd missed some on my first go, but not that many. 😅

@tdhock, two (or more) of those topics aren't functions and should instead be vignettes. They are `translate_qplot_ggplot` and `translate_qplot_lattice` [1]. They come from `ggplot2`, though they and `qplot()` have since been deleted.  Should I leave them as topics, change them into vignettes, or delete them outright? Or do something else? Thanks. :>

[1] https://github.com/animint/animint2/blob/f99ea2cb4e011f80026e1d6fbdf32f2fa41eaf2b/R/translate-qplot-ggplot.r & https://github.com/animint/animint2/blob/f99ea2cb4e011f80026e1d6fbdf32f2fa41eaf2b/R/translate-qplot-lattice.r

**Comment by ampurr** [comment]
Status update: All 50 aforementioned topics have been added. Unfortunately, this does mean I'll need to re-organize the functions again. Previously, the categories were dissatisfactory. Now, they're dissatisfactory and also messy and hard to navigate and probably aesthetically displeasing.

For example, one section looks like this:

``` yaml
- title: "Animint & Animint Helpers"
  desc: >
    ...
- contents:
  - starts_with("animint")
  - addShowSelectedForLegend
  - addSSandCSasAesthetics
  - starts_with("check")
  - colsNotToCopy
  - starts_with("get")
  - issueSelectorWarnings
  - make_bar
  - make_tallrect
  - make_widerect
  - make_text
  - merge_recurse
  - newEnvironment
  - parsePlot
  - pt.to.lines
  - renderAnimint
  - saveChunks
  - selectSSandCS
  - setPlotSizes
  - split_recursive
  - switch_axes
  - varied.chunk
```

I think I'll probably break that into two subsections at minimum.

`translate_qplot_ggplot` and `translate_qplot_lattice` have been placed under `title: internal`, which just means that they're out of the public eye till I get some more guidance about what to do about them. 🐈

**Comment by ampurr** [comment]
Status update: Working through `_pkgdown.yml`. I think the functions have mostly been reorganized, and I'm writing and editing the descriptions for the different sections. Feeling pretty good about that. A little uncertain about the "Animint Helpers" section, since there's so many functions in it [1]. But maybe sections can have lots of functions in them and that's okay.

[1] It looks like this:

``` yaml
- subtitle: "Animint Helpers"
  desc: >
    ...
- contents:
  - addShowSelectedForLegend
  - addSSandCSasAesthetics
  - starts_with("check")
  - colsNotToCopy
  - starts_with("get")
  - issueSelectorWarnings
  - make_bar
  - make_tallrect
  - make_widerect
  - make_text
  - merge_recurse
  - newEnvironment
  - parsePlot
  - pt.to.lines
  - renderAnimint
  - saveChunks
  - selectSSandCS
  - setPlotSizes
  - split_recursive
  - switch_axes
  - varied.chunk
```

**Comment by ampurr** [comment]
Addendum: Thoughts on adding [your YouTube video](https://youtu.be/VwHBiopG7b8) to the README? Didn't realize this video existed till now. :O

**Comment by ampurr** [comment]
Also, lemme know if you want a News section on the website. If so, I'll convert `NEWS` to `NEWS.md` and make some changes to the formatting.

**Comment by ampurr** [comment]
Status update: `_pkgdown.yml` is complete and also hopefully full of useful comments. The resulting reference page looks good. :>

**Comment by tdhock** [comment]
all this sounds good but I still am unable to preview the web site, even after clicking the links you provided in comments above. if you can not provide a link to preview, can you please provide instructions about how you are viewing the site, that I could follow?

**Comment by ampurr** [comment]
Hey, Toby. :>

The links are to the code or markup that'll produce the website, not to the website itself. Should they normally allow you to preview the website? I'm less familiar with GitHub than you are, so please let me know if I've messed something up re: previewing the website. 😅

If you wanna wait, I'm planning on throwing the website online today. And you can view it from there and suggest changes.

If you don't, you can build the website yourself by:

1. Cloning the website branch ([reference-website-yay](https://github.com/animint/animint2/tree/reference-website-yay))
2. `install.packages(pkgdown)`
3. `pkgdown::build_site()`

Hope this helps. Let me know if I've misunderstood your message. 😺

**Comment by ampurr** [comment]
Oh, I should clarify the `pkgdown` website situation.

How do I build a website and ask GitHub to throw it online for me? There are two methods:

- Build the website with `pkgdown::build_site()`, which will produce a bunch of HTML files in a `/docs` directory. Then commit that directory and its contents to git.
- Build the website, but _don't_ commit the directory to git. Instead, set up a GitHub Action to build it for you.

To be honest, I've never gotten the second method to work before on my personal websites. But I'm probably gonna try it anyway. If it fails, I'll just go to the first method.

Let me know if you have any preferences or if I should just do whatever. Thank you. :>

**Comment by ampurr** [comment]
Here it is: https://animint.github.io/animint2/

**Comment by ampurr** [comment]
Note to self: The rendered HTML does include `<script>` tags that refer to what `animint2` should've outputted. Some of the output doesn't seem to exist. I wonder why. `pkgdown` outputs static plots from `animint2`...

**Comment by ampurr** [comment]
Note to self: `pkgdown` writes the JavaScript for the first R code block, but none of the other code blocks. Untested wild hypothesis—`pkgdown` loads code blocks independently from one another, unlike how most R Markdown and Quarto files do it.

**Comment by ampurr** [comment]
Minor status update: `pkgdown` exhibits a bug where `![alt text](image_file)` causes the alt text to also appear in the fig caption. I've bypassed it by replacing the Markdown image syntax with an HTML `<img>` instead.

Planning on creating a reproducible example and filing a bug report with the `pkgdown` people.

**Comment by ampurr** [comment]
Also! Getting pretty close to finishing this up. Lots of little stuff left, but we have a working website and all. So the big stuff seems finished. Though let me know if you disagree, of course.

Checklists to keep myself organized:

### Fix Alt Text Redundancy so it Stops being Redundant

- [x] Fix alt text–fig caption duplication problem.
- [ ] Create reproducible example. See [here for an example](https://github.com/r-lib/pkgdown/blob/c9206802f2888992de92aa41f517ba7812f05331/.github/CONTRIBUTING.md#package-reprexes) of how to do it.
- [ ] Create issue on [r-lib/pkgdown](https://github.com/r-lib/pkgdown/issues). Link reproducible example.

### Animints Need to Show Up in the Quick Start Guide

- [ ] Take a look at [Debugging from _Advanced R_](https://adv-r.hadley.nz/debugging.html).
- [ ] Figure out why `pkgdown` renders code chunks 1-6, but fails to render any chunks afterward. Chunk 6 was the first animint thing.
- [ ] Figure out why Chunk 6—`animint(cute_1)`—doesn't produce the interactive JavaScript data viz that it should.
- [ ] Fix the problem. Possible solution: render everything manually and force it onto the page with `<script>`s. But it's obviously better if I can get `pkgdown` to do it instead.

### Writing is Important and so is Documentation

- [ ] Look over and edit `README.md`.
- [ ] Look over and edit the quick start guide.
- [ ] Finish up the website setup page.
- [ ] Look over and edit the website setup page.

### I Like it When Websites are Pretty

- [ ] Touch up the website's CSS. That includes: link colors, padding between sections, and font size.
- [ ] Create an interactive hexagon badge for the Animint2 package.

### Stop Pretending to be a Digital Hermit

- [ ] Concatenate the questions you've asked in this thread. Discard the irrelevant ones and present the relevant ones to Toby.
- [ ] Get Toby's feedback on the website as a whole.

### You Have Other Issues

- [ ] Close issue #100. Add the TODOs.
- [ ] Close issue #96. Look into it. If that's not possible, close.

**Comment by ampurr** [comment]
Note to self: the code chunks are now being rendered with JavaScript and everything. But the JS isn't rendering on the page.

**Comment by ampurr** [comment]
Status update: Made some aesthetic changes to the reference site. I've changed the link color to match the one on your personal website, increased spacing where I thought it would help, and switched the typeface family to Roboto [1].

I think that's improved both the appearance and readability of the website a lot, but you're the judge, not me. Let me know what you think and if I should revert or make any additional changes.

[1]: Most prominently Roboto Condensed. The Roboto family isn't my favorite. But it looks better than the system-ui default. I've also been using Roboto in my other GSOC materials.

**Comment by tdhock** [comment]
looks great overall thanks for making the web site!
On https://animint.github.io/animint2/articles/quick-start.html I see you have a block of What's happening with the code here? then 1. 2. 3. 4. 5. explaining the lines of code above, and that is fine, but I saw a talk by Mine-Cetankaya Rundel last week at JSM, which explains how quarto code annotation https://quarto.org/docs/authoring/code-annotation can be used to actually put the 1 2 3 4 5 next to the relevant lines of code, so that is something you can consider implementing if you have time/interest.
Also on that page, the animints are not being rendered. Did you check the right click menu options "view page source" or "Inpect" in your web browser? That shows the generated HTML and could be useful for debugging.
The integration with Rmd is implemented in knit_print.animint, https://github.com/animint/animint2/blob/master/R/z_knitr.R in case you need to change/update anything on our side. Can you please link the pkgdown issue here if/when you create one?

**Comment by ampurr** [comment]
Ah-ha. From the [pkgdown reference site](https://pkgdown.r-lib.org/reference/build_articles.html): "`build_articles()` renders each R Markdown file underneath `vignettes/` and saves it to `articles/`. That explains why it's generating the animints in `vignettes/`.

**Comment by ampurr** [comment]
### Minor Status Update

I've moved the quick start guide to a "Get started" section on the nav-bar.

On a related note, I got the animints working... Is what I wanna say. What actually happened is that renaming the file and moving it to its own section somehow got the animints to work. So it's not accurate to say that _I_ got the animints working.

Why did this work? I can't think of any plausible hypotheses. This is so strange.

**Comment by ampurr** [comment]
### Status Update

I've managed to hack together code annotations in the comments. Except for the hash sign, I think it looks a little bit better than Quarto's solution. Since I had no control over the HTML of the codeblock, I'm pretty proud of my workaround solution. (Maybe I'll be less proud when I wake up. 😛)

Due to [certain limitations in CSS's `::marker` pseudo-element](https://developer.mozilla.org/en-US/docs/Web/CSS/::marker#allowable_properties), getting the writing explaining the code annotations to look like Quarto's solution is harder. Currently it's a WIP.

Quarto's solution is to use `<dt>` and a CSS grid. I'm trying to get an ordered list to work instead.

**Comment by ampurr** [comment]
## Status Update

Got the CSS for the explanations for the code annotation working. Gonna throw together a PR to see how my commits test and then I'm gonna sleep. 😴

**Comment by ampurr** [comment]
Status update: I moved `website_setup.Rmd` to my GSOC subdomain—at some point I realized it wasn't necessary. Replacing it with a shorter document on maintaining the website.

**Comment by ampurr** [comment]
@tdhock, I moved it to its own space on the navbar. It's here now: https://animint.github.io/animint2/articles/animint2.html

**Comment by ampurr** [comment]
### Status Update

- Finished the website maintenance/debugging internal documentation. It's `readme_website.md` and [can be read here](https://github.com/animint/animint2/blob/reference-website-yay/readme_website.md).
- Converted `NEWS` to `NEWS.md` and made a couple of minor changes. This lets me put it on the website as the Changelog. If you don't want that, lemme know and I'll revert. :>
  - All headings are now Markdown headings, e.g. `# Changes in...`.
  - All headings have had "version" removed, e.g. "Changes in 2023.6.11" instead of "# Changes in version 2023.6.11."
  - No other changes.

**Comment by tdhock** [comment]
Hi thanks for sharing. Looks great overall. Glad you could get the line numbering and animint rendering to work. Can you please document the solution to those issues somewhere? (maybe on the wiki?)
Here are a few minor comments/suggestions:

* the plots with `geom_smooth` seem to be more complicated than necessary. For an intro / quick start, I think it would be better to just stick to the `geom_point`, first static then animint.
* using the pipe seems more complicated than necessary. Usually pipes are nice and clear when there are several (more than one) different nested function calls, but on that page the examples have max 1 pipe, so I would suggest re-writing the following
 
```r
cute_present <- cute_1 |> 
  animint(first = list(Cat = c("Archibald", "Muffin")))
```
as
```r
cute_present <- animint(
  cute_1, 
  first=list(Cat = c("Archibald", "Muffin"))
```

For the animated plot, there is a warning,
```
## Warning in issueSelectorWarnings(meta$geoms, meta$selector.aes, meta$duration):
## to ensure that smooth transitions are interpretable, aes(key) should be specifed
## for geoms with showSelected=Day, problem: geom1_point_associations
```
can you please specify `aes(key)` ? I think it should be `key=Cat`, right?

Also can you add a section which adds a second plot with clickSelects? maybe plot some aggregate vs Day, with `geom_tallrect(aes(clickSelects=Day))` ?

Also can you please move the data/aes to inside the geom? (this is more consistent with the other tutorials, and more generally useful, since in animints, different layers tend to have different data sets and aes) For example

```r
cute_1 <- meowtrics |>                                    
  ggplot() +                                              
  geom_point() +
  aes(x = Day, y = Cuteness, group = Cat, color = Cat)
```
could be changed to 
```r
cute_1 <- ggplot()+
  geom_point(aes(
    x = Day,
    y = Cuteness,
    group = Cat, 
    color = Cat),
    data = meowtrics)
```
or, alternatively, if you think the current version has some kind of advantage, can you please show both versions, discuss how they give the same result, and also discuss the relative advantage/disadvantage of each approach/syntax?

Thanks!

**Comment by ampurr** [comment]
Thanks for the thorough feedback! I'll make the changes and then get back to you. :>

**Comment by ampurr** [comment]
## Changes to the Quick Guide

### Refactoring the Code
- [ ] Remove all pipes. Toby says they're needlessly complex.
- [ ] Move the data and aes inside the geom. Toby says it's for consistency and usefulness in `animint2`.
- [ ] Specify the animint's `aes(key)` to remove the warning.

### Substantial Changes
- [ ] Add a second plot with `clickSelects`.
- [ ] Remove the plots with `geom_smooth`. Toby said it's too complicated for an intro.
- [ ] Add more code annotations to help the reader.
- [ ] Add alt text to the plots.
- [ ] Document how I got the code annotations and animint rendering to work.

**Comment by ampurr** [comment]
### Status Update

I've finished most of the changes. The quick start guide on the website is currently a mess, though.

Currently I'm stuck on the linked animints (with `geom_tallrect`). Every time I resolve an error another pops up. But I think I've figured it out—I think I need to remove redundant days in `meowtrics$Day`. I'll test it out when I get up.

**Comment by tdhock** [comment]
looks like good progress.
one correction "The specific cats must also be in a list" -> actually should be character vector.
right, for the section about clickSelects you probably need to use something like `unique(data.frame(Day=meowtrics$Day))` to create a data frame of unique days.

**Comment by ampurr** [comment]
Thanks for the correction! I'll fix that.

Yup! That did it, thanks.

Still working on the linked animints. Everything seems right, but it produces a static data viz instead of an animint. Still trying to figure that out.

**Comment by tdhock** [comment]
> Still working on the linked animints. Everything seems right, but it produces a static data viz instead of an animint. Still trying to figure that out.
rather than linked animints do you mean linked plots in a single animint? That is what it should be, there is no way to link two different animints. if you share your code, I can help

**Comment by ampurr** [comment]
Yes, I meant linked plots in a single animint. I assumed that plots and animints had a one-to-one correspondence (cuz I thought "animint" referred to interactive data viz). What exactly is an animint? 😅

Thanks—I really appreciate the offer. 😄 I actually just figured it out. After hours debugging and re-reading thinking that my code basically looked the same as the code in the manual, I realized that the problem was basically a typo. Whoops.

Now the linked plots work. Thanks again! 🐈

**Comment by ampurr** [comment]
### Status Update

I've finished with the linked plots and fixed the mistake with `c()`. I'm sure there are typos and awkward bits of writing, so I'm gonna go over the quick start guide again tomorrow.

If there are any other minor changes I should make, please let me know, and I'll make them.

If there are any major changes I should make, also let me know, and I'll make them after GSOC ends. The work product submission deadline is coming up and I wanna dedicate at least most of a day to it.

In other news, `pkgdown` footnotes apparently rely on a JS library called Popper. Also, they're not working on the website and I can't control how the HTML comes out [1]. I might just replace them with `<aside>`s [2].

[1] I tried a bunch of things but I'm real tired and don't wanna list them out.
[2] Unlike JavaScript, HTML has never betrayed me. I love you, HTML. 🐈‍⬛

**Comment by ampurr** [comment]
Status update: I've replaced the Markdown footnotes (which weren't working and relied on an external JS lib) with HTML ones.

**Comment by ampurr** [comment]
Quick guide status update: I've edited my writing and cleaned up typos.

**Comment by tdhock** [comment]
Hi again thanks for the updates. things are improving for sure.
the scrolling bug is apparent #92 so you may consider discussing a work around (click Hide Selection Menus Button in animated_associations)
by the way, variable names like `animated_associations` are preferable to `cute_plot_4` (names are easier to understand, numbers in variable names should be avoided) so you you please update the variable names?
also for `cute_plot_4` it would be easier to understand if you add 

* a `geom_text(showSelected="Kind")` so that the user can see what Kind is currently selected
* `geom_point(color_off="transparent", color="black")` or something else to provide greater emphasis about what points are selected (that also offers an opportunity for teaching about `color_off`)

I would suggest changing the title "Advanced: Linked Plots" by removing the word "Advanced" -- the normal usage of animint is to have at least two linked plots which use clickSelects and showSelected. Using only one plot is typically only in teaching. 
Also for that plot it works fine, glad you got it to work. Please consider adding another animint afterwards where you use aligned axes in a single plot, rather than two different plots (y axis is Cuteness). Aligned axes is usually preferable when possible, to (1) save space, (2) avoid repeating legends, and (3) emphasize the common axis. This technique is discussed in https://rcdata.nau.edu/genomic-ml/animint2-manual/Ch02-ggplot2.html#multi-panel

In Conclusion can you please add a link to the Animint2 Manual? Your quick start guide is roughly equivalent to Chapters 2-4, so I would mention something like "You can see another presentation of the same material in more depth in Chapters 2-4 of the Animint2 Manual, https://rcdata.nau.edu/genomic-ml/animint2-manual/Ch02-ggplot2.html#multi-panel"

Thanks again for your great work.

**Comment by ampurr** [comment]
Sure thing. I'll make those changes and let you know when it's done. :>

Will probably use an `<aside>` to discuss the bug.

**Comment by ampurr** [comment]
New checklists:

### Changes to the Quick Start Guide
- [ ] Write `<aside>` discussing Firefox bug and the hide selection menus solution.
- [ ] Remove variable names with numbers and replace them with words.
- [ ] Remove the word "advanced" from "Advanced: Linked Plots."
- [ ] Add a new section where everything is in one plot instead of two. (From Toby: [see the manual](https://rcdata.nau.edu/genomic-ml/animint2-manual/Ch02-ggplot2.html#multi-panel).)
- [ ] Add a link to the animint2 Manual in the conclusion. Mention chapters 2 to 4.
- [ ] Document how I got the code annotations and animint rendering to work.

### Address Alt Text Redundancy
- [ ] Create reproducible example. See [here for an example](https://github.com/r-lib/pkgdown/blob/c9206802f2888992de92aa41f517ba7812f05331/.github/CONTRIBUTING.md#package-reprexes) of how to do it.
- [ ] Create issue on [r-lib/pkgdown](https://github.com/r-lib/pkgdown/issues). Link reproducible example.

### Aesthetic Improvements
- [ ] Create an interactive hexagon badge for the Animint2 package.

**Comment by ampurr** [comment]
### Status Update

1. Wrote aside and wrote some CSS to have its background complement the link colors.
2. Added a link to the animint2 Manual in the conclusion.
3. Made other relatively small changes.

**Comment by tdhock** [comment]
hi  @ampurr is this the token you were using to push the gh-pages web site from your own personal computer? or?
![image](https://github.com/animint/animint2/assets/932850/e3a383b0-e282-4a71-98f2-33b03af8d1a3)
I see on your docs https://github.com/animint/animint2/blob/master/readme_website.md#why-dont-you-just-use-github-actions that you did not know how to setup auto deploy in github actions, so I guess that means this token was never necessary for github actions.

**Comment by ampurr** [comment]
Hey, @tdhock. :>

Yeah, this is the token I used to push the website from my personal computer. People need to renew every year or so, which is kinda annoying. I think the classic token can be permanent. Having never used GitHub Actions, I dunno if this token is necessary for GitHub Actions or not. 😅

It's possible that I might one day figure out auto deploy for this site. But I've still got a [few changes](https://github.com/animint/animint2/issues/97#issuecomment-1696706789) I need to make to the site (which you're requested and which I've been terribly slow at addressing). So, no promises on the auto deploy front.

**Comment by ampurr** [comment]
## For Clarity

Well, when I update the website, it does activate a GitHub workflow, which I think is a subset or synonym of a GitHub action. I'm not clear on that, since I've only read a limited subset of the GitHub documentation. If you're using "GitHub action" in the sense that GitHub starts a workflow in the background when I update the site, then this token _is_ necessary for at least some GitHub actions.

I hope my response is relevant to your guess about this token being necessary or unnecessary for GitHub Actions. Sorry if it isn't. 😺
